### PR TITLE
wrap lists of variants for #1258

### PR DESCRIPTION
**Pull request by davidcarlisle**
Prevent lists of variants being too long, #1258 

Two changes from the code in the issue:
I went for measuring an hbox rather than playing with `\prevgraf` as it seemss simpler. Also ensured a trailing TF stayed with the )

The cut off is set at an arbitrary `.4\columnwidth`. I did consider making that length or the scale factor a documented setting you could set in the cfg but I'm not sure we want to promise a stable interface here so not done at this time.

### CODE REVIEW

**Review by josephwright — APPROVED** [review]

**Review by josephwright — APPROVED** [review]

File: l3kernel/CHANGELOG.md [hunk 1]
```diff
@@ -15,9 +15,12 @@ this project uses date-based 'snapshot' version identifiers.
 - Several missing `\tex_<name>:D` primitives from LuaTeX and (up)TeX
 
 ### Changed
+
 - `\tex_protrudechars:D` now defined for XeTeX (to `\XeTeXprotrudechars`)
 - Allow for optional `v` in version string argument to
   `\ProvidesExpl...`
+-  Wrap long lists of variants in `l3doc` display of functions (issue \#/1258)
```
Comment (muzimuzhi) [inline l3kernel/CHANGELOG.md:22]:
```suggestion
-  Wrap long lists of variants in `l3doc` display of functions (issue \#1258)
```

File: l3kernel/l3doc.dtx [hunk 2]
```diff
@@ -2751,11 +2751,27 @@ and all files in that bundle must be distributed together.
     \int_compare:nTF { \seq_count:N \g_@@_variants_seq == 1 }
       { \seq_use:Nn \g_@@_variants_seq { } }
       {
+       \hbox_set:Nn\l_tmpa_box
+         { \seq_use:Nn \g__@@_variants_seq { \textrm|\nolinebreak[2]} }
```
Comment (muzimuzhi) [inline l3kernel/l3doc.dtx:2755]:
```suggestion
      {
        \hbox_set:Nn \l_tmpa_box
          { \seq_use:Nn \g__@@_variants_seq { \textrm| \nolinebreak[2] } }
```

File: l3kernel/l3doc.dtx [hunk 3]
```diff
@@ -2751,11 +2751,27 @@ and all files in that bundle must be distributed together.
...
+             {\raggedright
+               \hbox_unpack_drop:N\l_tmpa_box
+               \textrm)
+               \bool_if:NT #2 { \@@_typeset_TF: }
+             }
+          }
+          { \hbox_unpack_drop:N\l_tmpa_box
+             \textrm)
+             \bool_if:NT #2 { \@@_typeset_TF: }
+          }
```
Comment (muzimuzhi) [inline l3kernel/l3doc.dtx:2773]: Not sure if you like this indent and spacing
```suggestion
        \dim_compare:nNnTF { \box_wd:N \l_tmpa_box } > { .4\columnwidth }
          {
            \parbox[t]{.4\columnwidth}
              {
                \raggedright
                \hbox_unpack_drop:N \l_tmpa_box
                \textrm)
                \bool_if:NT #2 { \@@_typeset_TF: }
              }
          }
          {
            \hbox_unpack_drop:N \l_tmpa_box
            \textrm)
            \bool_if:NT #2 { \@@_typeset_TF: }
          }
      }
```

**Review by muzimuzhi — COMMENTED** [review]
Mostly minor styling suggestions.

**Comment by davidcarlisle** [comment]
> Mostly minor styling suggestions.

yes thanks, I'll clean up before merging.

It's almost as If I got my tex coding style in simpler times when all vowels got replaced by `@` and you didn't add any cosmetic white space to keep the byte size of the file down:-)

**Review by FrankMittelbach — APPROVED** [review]

File: l3kernel/l3doc.dtx (hunk 3 above)
Comment (davidcarlisle) [reply l3kernel/l3doc.dtx:2773]: I think this one's got an extra } at the end, I'll apply locally

File: l3kernel/CHANGELOG.md [hunk 4]
```diff
@@ -15,9 +15,12 @@ this project uses date-based 'snapshot' version identifiers.
 - Several missing `\tex_<name>:D` primitives from LuaTeX and (up)TeX
 
 ### Changed
+
```
Comment (muzimuzhi) [inline l3kernel/CHANGELOG.md:18]: It seems by convention this empty line is surplus.

File: l3kernel/l3doc.dtx (hunk 3 above)
Comment (muzimuzhi) [reply l3kernel/l3doc.dtx]: Ya you're right. I pasted too much 😅 (from a locally modified copy of the whole `\int_compare:nTF { \seq_count:N \g_@@_variants_seq == 1 }` block).

File: l3kernel/CHANGELOG.md (hunk 4 above)
Comment (davidcarlisle) [reply l3kernel/CHANGELOG.md:18]: sorry I just missed that before merge, but the changelog will get further edits anyway before it goes out. Thanks for your review

File: l3kernel/l3doc.dtx (hunk 3 above)
Comment (davidcarlisle) [reply l3kernel/l3doc.dtx]: no problem, this got done as 101dbb4514a9da07b447eec1610afb721e80cfc9

**Pull request merged**